cmd/containerd-shim-runhcs-v1/delete.go

```diff
@@ -35,27 +35,24 @@ The delete command will be executed in the container's bundle as its cwd.
 
 		// Attempt to find the hcssystem for this bundle and terminate it.
 		if sys, _ := hcs.OpenComputeSystem(idFlag); sys != nil {
+			defer sys.Close()
 			if err := sys.Terminate(); err != nil {
-				if hcs.IsPending(err) {
-					const terminateTimeout = time.Second * 30
-					done := make(chan bool)
-					go func() {
-						if werr := sys.Wait(); err != nil {
-							fmt.Fprintf(os.Stderr, "failed to wait for '%s' to terminate: %v", idFlag, werr)
-						}
-						done <- true
-					}()
-					select {
-					case <-done:
-					case <-time.After(terminateTimeout):
-						sys.Close()
-						return fmt.Errorf("timed out waiting for '%s' to terminate", idFlag)
+				fmt.Fprintf(os.Stderr, "failed to terminate '%s': %v", idFlag, err)
+			} else {
+				ch := make(chan error, 1)
```

Comment:
What is the point of a buffered channel for a select?

Reply:
In the timeout case, we will never read the message, so if we don't buffer it then the goroutine will block forever.